deploy/charts/emqx/templates/service-monitor.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if and (.Values.metrics.enabled) (eq .Values.metrics.type "prometheus") }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "emqx.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ include "emqx.name" . }}
+    helm.sh/chart: {{ include "emqx.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+  {{- if .Values.service.annotations }}
+  annotations:
+    {{ toYaml .Values.service.annotations | indent 4 }}
+  {{- end }} 
+spec:
+  endpoints:
+  - interval: 10s
+    port: mgmt 
+    scheme: http
+    path: /api/v4/emqx_prometheus
```

Comment:
This relies on the `emqx_prometheus` plugin for emqx, which we should probably turn on when we enabled metrics